meorg_client/client.py

```diff
@@ -127,51 +172,54 @@ def login(self, email: str, password: str):
         }
 
         # Call
-        response = self._make_request("post", endpoint=endpoints.LOGIN, data=login_data)
+        response = self._make_request(
+            method=mcc.HTTP_POST,
+            endpoint=endpoints.LOGIN,
+            json=login_data,
+            return_json=True,
+        )
 
         # Successful login
-        if response.status_code == 200:
+        if self.last_response.status_code == 200:
             auth_headers = {
-                "X-User-Id": response.json()["userId"],
-                "X-Auth-Token": response.json()["authToken"],
+                "X-User-Id": response["data"]["userId"],
+                "X-Auth-Token": response["data"]["authToken"],
             }
 
             self.headers.update(auth_headers)
 
         # Unsuccessful login (technically this will have already failed)
         else:
-            raise Exception("Login failed")
+            raise RequestException(
+                self.last_response.status_code, self.last_response.text
+            )
 
     def logout(self):
-        """Log the user out. Likely not necessary, can just let sessions expire.
-
-        Returns
-        -------
-        dict or requests.Response
-            Response from ME.org.
-        """
-        response = self._make_request(mcc.HTTP_POST, endpoint=endpoints.LOGOUT)
+        """Log the user out. Likely not necessary, can just let sessions expire."""
+        response = self._make_request(
+            method=mcc.HTTP_POST, endpoint=endpoints.LOGOUT, return_json=False
+        )
 
         # Clear the headers.
-        self.headers = dict()
-
-        return response
+        if response.status_code == 200:
+            self.headers.pop("X-User-Id", None)
+            self.headers.pop("X-Auth-Token", None)
```

Comment:
Maybe have an error case if request wasn't successful (same for `login`). Also checking for 200 response code exactly here (I think creating a smaller function like `success()` would be good?

Reply:
Thanks, I think the class will trigger something anyway with the internal request method - but I will check.

Reply:
Yes, an exception is raised in `_make_request()` which bubbles up here and constitutes an error.

login/logout are the only time we look for the explicit 200 code. If we end up with many more we will look at implementing your suggestion but at the moment it is likely unnecessary abstraction.